modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("Checks config modal opens when config button clicked", async () => {
+      render(<TemplateDetail />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getByTestId("template-detail.add-field");
+      act(() => templateEdit.click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.field-type-select"),
+        ).toBeInTheDocument();
+      });
+
+      const typeSelectElement = screen.getByTestId(
+        "template-detail.field-type-select",
+      );
+
+      await userEvent.selectOptions(typeSelectElement, "number");
+      expect(typeSelectElement).toHaveValue("number");
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      const configButton = screen.getByTestId(
+        "template-detail.edit-field-config",
+      );
+      act(() => configButton.click());
+
+      const modal = await screen.findByTestId("template-detail.field-modal");
+      expect(modal).toBeInTheDocument();
+    });
+
+    it("Checks select config modal is populated", async () => {
+      render(<TemplateDetail />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("template-detail.edit");
+      act(() => templateEdit[1].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.field-type-select"),
+        ).toBeInTheDocument();
+      });
+
+      const typeSelectElement = screen.getByTestId(
+        "template-detail.field-type-select",
+      );
+      expect(typeSelectElement).toHaveValue("select");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      const configButton = screen.getByTestId(
+        "template-detail.edit-field-config",
+      );
+      act(() => configButton.click());
+
+      const modal = await screen.findByTestId("template-detail.field-modal");
+      expect(modal).toBeInTheDocument();
```

Comment:
This is used 5 times, we can create a helper function in this describe for this (is a way to make shorter the patterns that repeat a lot on the test cases. `openModal()`